classes/Authentication/SAML/XDSamlAuthentication.php

```
@@ -132,14 +148,18 @@ public function getLoginLink()
                 );
             }
             return array(
-                    'url' => $this->_as->getLoginUrl(),
+                    'url' => $this->_as->getLoginUrl($returnTo),
                     'organization' => $orgDisplay,
                     'icon' => $icon
                 );
         } else {
             return false;
```

probably make this throw an exception rather than mixed return